bumble/hci.py

```diff
@@ -5148,8 +5161,8 @@ def from_bytes(cls, packet: bytes) -> HCI_Event:
         elif event_code == HCI_VENDOR_EVENT:
             # Invoke all the registered factories to see if any of them can handle
             # the event
-            if cls.vendor_factory:
-                if event := cls.vendor_factory(parameters):
+            for vendor_factory in cls.vendor_factories:
+                if event := vendor_factory(parameters):
                     return event
 
             # No factory, or the factory could not create an instance,
```

Comment:
Should here return HCI_Vendor_Event instead of HCI_Event?

Reply:
Yes. Done. I've also added a test for it.